src/groups/mqb/mqbblp/mqbblp_rootqueueengine.cpp

```diff
@@ -798,6 +797,16 @@ mqbi::QueueHandle* RootQueueEngine::getHandle(
 
         d_queueState_p->adopt(iter->value());
         upstreamSubQueueId = iter->value()->upstreamSubQueueId();
+
+        if (!iter->value()->isAuthorized()) {
+            if (iter->value()->authorize()) {
```

Comment:
If `authorize()` call returns false here, is this a reason for concern?

Reply:
Consider the race: Domain is reconfigured but VirtualStorage is not added yet.
`setQueue` is not called yet.
`isAuthorized()` is `false` and we need to `authorize()`

Another case is when App is not registered.  

To be refactored when we add Dynamic Apps